code/weapon/weapons.cpp

```diff
@@ -7840,3 +7920,40 @@ const char* weapon_info::get_display_string() {
 bool weapon_info::has_alternate_name() {
 	return alt_name[0] != '\0';
 }
+
+// Given a weapon, figure out how many independent locks we can have with it.
+int weapon_get_max_missile_seekers(weapon_info *wip)
+{
+	int max_target_locks;
```

Comment:
you could return the results immediately instead of storing them in the temporary variable first.